addons/l10n_in_purchase_stock/models/stock_move.py

```diff
@@ -9,8 +9,15 @@ class StockMove(models.Model):
     def _l10n_in_get_product_price_unit(self):
         self.ensure_one()
         if line_id := self.purchase_line_id:
+            company_id = line_id.company_id
             if qty := line_id.product_qty:
-                return line_id.price_subtotal / qty
+                return line_id.currency_id._convert(
+                    line_id.price_subtotal / qty,
+                    company_id.currency_id,
+                    company_id,
+                    fields.Date.today(),
```

Comment:
This means that if you call the method another day it will give another answer.  Is that how it is supposed to work?

Reply:
Oops my bad, `line_id.order_id.date_order` should do the job